### PR TITLE
Camera console update

### PR DESCRIPTION
**Pull request by Markolie**
DNM until https://github.com/ParadiseSS13/Paradise/pull/5540 is merged, as it relies on that pull request. It won't compile until then. Once that PR is merged I'll test this.

This refactors the security console camera so camera changes are hopefully instant and more reliable. These changes are based off of /tg/.

Fixes https://github.com/ParadiseSS13/Paradise/issues/5032.
Fixes https://github.com/ParadiseSS13/Paradise/issues/4890.

🆑 Markolie
tweak: The camera console has been refactored so there's no longer a delay when switching cameras.
fix: The camera console should no longer lag for several seconds.
/🆑

### CODE REVIEW

**Comment by Markolie** [comment]
Derp, looks like the reset_perspective changed weren't included in the status update PR, but instead in https://github.com/ParadiseSS13/Paradise/pull/5540. I'll have to wait for that now...

**Comment by Crazylemon64** [comment]
Oh! Oh! I'll bump up the priority of porting over `reset_perspective` from #5540, I was unaware that that was a blocker!

**Comment by Markolie** [comment]
Awesome, thank you @Crazylemon64!

**Comment by Crazylemon64** [comment]
ping!
#5840 
---
Replace uses of `reset_perspective` with `reset_view`

**Comment by Markolie** [comment]
Reminder to self: when testing this, see if removing the auto-update on the console works as a fix for https://github.com/ParadiseSS13/Paradise/issues/4890.

**Comment by Markolie** [comment]
This is good to go! It should also solve the camera freezing issue: I got rid of the auto update on the camera console because the ui_data proc is quite heavy, and there's no point in having it auto update.

File: code/game/machinery/computer/camera.dm [hunk 1]
```diff
@@ -1,94 +1,119 @@
...
+	available_networks["CentComm"] =          list(access_cent_security,access_cent_commander)
+	available_networks["Thunderdome"] =       list(access_cent_thunder,access_cent_commander)
+	
+/obj/machinery/computer/security/Destroy()
+	if(watchers.len)
+		for(var/mob/M in watchers)
+			M.unset_machine() //to properly reset the view of the users if the console is deleted.
+	return ..()
+
+/obj/machinery/computer/security/check_eye(mob/user)
```
Comment (Fox-McCloud) [inline code/game/machinery/computer/camera.dm]: Shouldn't this be, something like:

```
/obj/machinery/computer/security/check_eye(mob/user)
	if((stat & (NOPOWER|BROKEN)) || user.incapacitated() || !user.has_vision())
		user.unset_machine()
		return
	if(!(user in watchers))
		user.unset_machine()
		return
	if(!watchers[user])
		user.unset_machine()
		return
	var/obj/machinery/camera/C = watchers[user]
	if(!can_access_camera(C, user))
		user.unset_machine()
		return	
	if(!C.can_use())
		user.unset_machine()
		return
	if(!issilicon(user))
		if(!Adjacent(user))
			user.unset_machine()
			return
	else if(iscyborg(user))
		var/list/viewing = viewers(src)
		if(!viewing.Find(user))
			user.unset_machine()
```

Comment (Markolie) [inline code/game/machinery/computer/camera.dm]: can_use_camera takes care of all of those checks.

Comment (Fox-McCloud) [inline code/game/machinery/computer/camera.dm]: Ahh, thank you. I looked at that, but it came up with something different. Thank you.

**Comment by Fox-McCloud** [comment]
LGTM. @Crazylemon64

**Pull request merged**